AlexaSkillsKit.Lib/Speechlet/SpeechletAsync.cs

```diff
@@ -180,6 +181,10 @@ public abstract class SpeechletAsync : ISpeechletAsync
             if (request == null) return;
 
             if (session.IsNew) {
+                if (session.Attributes == null)
```

Comment:
Currently all changes to Speechlet.cs and SpeechletAsync.cs have to done in sync. Also, consider moving added block outside `if (session.IsNew)` check, as we can't expect Alexa will always accept/preserve session attributes received in previous responses.